File: WaermepumpeSimulator/Interop/PlotlyInterop.cs
using System.Text.Json;
using Microsoft.JSInterop;

namespace WaermepumpeSimulator.Interop;

public class PlotlyInterop
{
    private readonly IJSRuntime _js;
    private static readonly JsonSerializerOptions JsonOpts = new() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };

    public PlotlyInterop(IJSRuntime js) => _js = js;

    public async Task NewPlot(string elementId, object traces, object layout)
    {
        await _js.InvokeVoidAsync("PlotlyInterop.newPlot", elementId,
            JsonSerializer.Serialize(traces, JsonOpts),
            JsonSerializer.Serialize(layout, JsonOpts));
    }

    public async Task Purge(string elementId)
    {
        await _js.InvokeVoidAsync("PlotlyInterop.purge", elementId);
    }
}

File: WaermepumpeSimulator/wwwroot/js/plotlyInterop.js
window.PlotlyInterop = {
    newPlot: function (elementId, traces, layout, config) {
        var el = document.getElementById(elementId);
        if (!el) return;
        Plotly.newPlot(el, JSON.parse(traces), JSON.parse(layout), config ? JSON.parse(config) : { responsive: true });
    },
    purge: function (elementId) {
        var el = document.getElementById(elementId);
        if (el) Plotly.purge(el);
    }
};
